sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitDecl.cs

```diff
@@ -472,7 +472,7 @@ private void VisitFunctionDecl(FunctionDecl functionDecl)
 
             var needsReturnFixup = isVirtual && NeedsReturnFixup(cxxMethodDecl);
 
-            if (!(functionDecl is CXXConstructorDecl))
+            if ((functionDecl is not CXXConstructorDecl) || (_config.OutputMode == PInvokeGeneratorOutputMode.Xml))
```

Comment:
I guess we should add a boolean to WriteReturnType indicating whether it's for a ctor or not to avoid this being baked into VisitDecl, that does nothing if that boolean is true on the C# backend.